packages/jest-worker/src/channel.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+'use strict';
+
+import EventEmitter from 'events';
+import serializer from 'jest-serializer';
+
+import type {Duplex} from 'stream';
+
+/**
+ * This class provides an thin layer to send objects over a socket by using the
+ * best available serialization mechanism, which is in turn provided by the
+ * jest-serializer module.
+ *
+ * In order to send, the payload is encoded as a buffer, and a 4-byte unsigned
+ * integer with the size is added at the front. When receiving, buffers are
+ * saved until there is enough information to parse a payload, then a "message"
+ * event is emitted.
+ */
+export default class extends EventEmitter {
+  _size: Buffer;
+  _stream: Duplex;
+
+  _receiveBuffers: Array<Buffer>;
+  _receiveBytes: number;
+  _receiveSize: number;
+
+  constructor(stream: Duplex) {
+    super();
+
+    this._size = Buffer.allocUnsafe(4);
```

Comment:
This is v5+ API, so technically we break Node 4 compat :/

Reply:
Should probably use https://github.com/LinusU/buffer-alloc-unsafe or something, then. Would be great to keep around node 4 support (unofficial though it is) until it's EOL in a couple of months